sunpy/visualization/imageanimator.py

```diff
@@ -878,3 +880,29 @@ def update_plot(self, val, line, slider):
         if val != slider.cval:
             line.set_ydata(self.data[self.frame_slice])
             slider.cval = val
+
+class ImageAnimatorWCS(ImageAnimator):
+    """
+    Same parameters as ImageAnimator with a extra parameter for WCS.
+    """
+    def __init__(self, data, wcs=None, image_axes=[-2, -1], axis_ranges=None, **kwargs):
+        self.wcs = wcs
+        super(ImageAnimatorWCS, self).__init__(data, image_axes=image_axes, axis_ranges=axis_ranges, **kwargs)
+
+    def _get_main_axes(self):
+        axes = self.fig.add_axes([0.1, 0.1, 0.8, 0.8], projection=self.wcs, slices=(0, 'y', 'x'))
+        axes.coords[2].set_ticks(exclude_overlapping=True)
```

Comment:
Same comment as above but relating to whether coords[2] or another element of coords.